pkg/utils/certificateUtils.go

```diff
@@ -17,33 +17,97 @@ package utils
 
 import (
 	"crypto/x509"
+	"io/fs"
 	"os"
+	"strings"
 
 	"path/filepath"
 
+	"github.com/deislabs/ratify/pkg/homedir"
 	"github.com/notaryproject/notation-go-lib/crypto/cryptoutil"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 )
 
 func GetCertificatesFromPath(path string) ([]*x509.Certificate, error) {
-
 	var certs []*x509.Certificate
+	fileMap := map[string]bool{} //a map to track path of physical files
+
+	path = ReplaceHomeShortcut(path)
 
 	err := filepath.Walk(path, func(file string, info os.FileInfo, err error) error {
-		if !info.IsDir() {
-			cert, certError := cryptoutil.ReadCertificateFile(file) // ReadCertificateFile returns empty if file was not a certificate
-			if certError != nil {
-				return certError
+
+		physicalFileInfo := info
+		physicalFilePath := file
+
+		if info == nil || err != nil {
+			logrus.Warnf("Invalid path '%v' skipped, error %v", file, err)
+			return nil
+		}
+
+		// In a cluster environment, each mounted file results in a physical file and a symlink
+		// check if file is a link and get the actual file path
+		if info.Mode()&os.ModeSymlink != 0 {
+
+			physicalFilePath, err = filepath.EvalSymlinks(file)
+
+			if err != nil {
+				logrus.Errorf("error evaluating symbolic link %v , error '%v'", file, file)
+				return nil
+			}
+
+			physicalFileInfo, err = os.Lstat(physicalFilePath)
```

Comment:
What if a symlink points to a symlink or other non-regular files?

Reply:
thanks @shizhMSFT, this is a good point. I think our options are handle a single level ( support how the files are mounted in the cluster) or try to resolve the link with with a maximum support depth (maybe 3 level ? ). do we know of a scenario where customer need to have multi level sym link?   I prefer adding support for the basic scenario for now, and handle multi level when needed. Let me what you think, thanks!

Reply:
added a unit test for two level symlink, the test result shows filepath.EvalSymlinks was able to resolve to the physical file. Please review. thanks.